python/tests/test_topology.py

```diff
@@ -3466,6 +3460,282 @@ def test_many_trees_recurrent_mutations_internal_samples(self):
+        self.assertEqual(list(tss.parent), [5, 7, 7, 8, 8])
+        self.assertEqual(list(tss.child), [1, 4, 5, 2, 7])
+        self.assertEqual(all(tss.left), 0)
+        self.assertEqual(all(tss.right), 1)
+
+    def test_multiple_trees_to_single_tree(self):
+        nodes = io.StringIO(self.nodes0)
+        edges = io.StringIO(self.edges0)
+        ts = tskit.load_text(nodes=nodes, edges=edges, strict=False)
+        tss = self.do_map(ts, ancestors=[5, 6])
+        self.assertEqual(list(tss.parent), [5, 5, 6, 6])
+        self.assertEqual(list(tss.child), [0, 1, 2, 3])
+        self.assertEqual(all(tss.left), 0)
+        self.assertEqual(all(tss.right), 1)
+
+    def test_multiple_trees_one_ancestor(self):
+        nodes = io.StringIO(self.nodes0)
+        edges = io.StringIO(self.edges0)
+        ts = tskit.load_text(nodes=nodes, edges=edges, strict=False)
+        tss = self.do_map(ts, ancestors=[9, 10])
+        self.assertEqual(list(tss.parent), [9, 9, 9, 9, 9, 10, 10, 10, 10, 10])
+        self.assertEqual(list(tss.child), [0, 1, 2, 3, 4, 0, 1, 2, 3, 4])
+        self.assertEqual(all(tss.left), 0)
+        self.assertEqual(all(tss.right), 1)
+
+    def verify(self, ts, sample_nodes, ancestral_nodes):
+        tss = self.do_map(ts, ancestors=ancestral_nodes, samples=sample_nodes)
+        # ancestors = list(set(tss.parent))
+        # Loop through the rows of the ancestral branch table.
+        current_ancestor = tss.parent[0]
+        current_descendants = [tss.child[0]]
+        current_left = tss.left[0]
+        current_right = tss.right[0]
+        for ind, row in enumerate(tss):
+            if row.parent != current_ancestor or\
+                    row.left != current_left or\
+                    row.right != current_right:
+                # Loop through trees.
+                for tree in ts.trees():
+                    if tree.interval[0] >= current_right:
+                        break
+                    while tree.interval[1] <= current_left:
+                        tree.next()
+                    # If there's just one child, check node is an ancestor
+                    # in the original tree.
+                    current_descendants = list(set(current_descendants))
+                    if len(current_descendants) == 1:
+                        self.assertTrue(
+                            tree.is_descendant(
+                                current_descendants[0],
+                                current_ancestor))
+                    # Otherwise, for all combinations of descendants, check that the
+                    # current ancestor has the mrca as a descendant.
+                    else:
+                        pairs = itertools.combinations(current_descendants, 2)
+                        for pair in pairs:
+                            mrca = tree.mrca(pair[0], pair[1])
+                            self.assertTrue(tree.is_descendant(mrca, current_ancestor))
```

Comment:
Another way to do both these checks (lines 3683-3694) would be to: for each of `current_descendants`, iterate up the tree until one of `ancestral_nodes` is reached, and check that is equal to `current_ancestor`. That would then check we've actually got the most recent ancestor for everyone instead of just an ancestor.

Reply:
After a bit of thought, yes, I think this is a better check. We were very fixated on the idea of doing a test involving MRCAs for some reason. I have changed this in the latest commit.